distribution/lib/Standard/Visualization/0.0.0-dev/src/Widgets.enso

```diff
@@ -27,9 +28,21 @@ get_widget_json value call_name argument_names uuids="{}" =
 
     read_annotation argument = Panic.catch Any handler=(log_panic argument) <|
         annotation = Warning.clear <| Meta.get_annotation value call_name argument
-        return_target err = err.payload.target
-        Panic.catch Not_Invokable handler=return_target
-            annotation value cache=cache
+        case annotation of
+            # The annotation is a function that is taking the 'self' value and _maybe_ a cache parameter.
+            f : Function ->
+                # We pass the self value and the cache to the function.
+                Panic.catch Not_Invokable (f value cache=cache) caught_panic->
+                    ## If the cache argument was not expected, we ignore the problem
+                       and return just the result of `f value` - held by
+                       the target of the Not_Invokable error.
+                    was_not_expecting_cache_argument = caught_panic.payload.cause == (No_Such_Argument.Error "cache")
```

Comment:
Question to myself: Why do I find this unreadable? Would following be better?

```ruby
f : Function ->
    Panic.catch Not Invocable (f value cache=cache) caught_panic -> case caught.panic.payload.cause of
       No_Such_Argument.Error "cache" -> caught_panic.payload.target
       _ -> Panic.throw caught_panic
```

Maybe, but `caught.panic.payload.cause` is still quite long...